pkgs/development/tools/build-managers/meson/default.nix

```diff
@@ -92,7 +95,24 @@ python3.pkgs.buildPythonApplication rec {
 
     # Do not propagate Python
     rm $out/nix-support/propagated-build-inputs
-  '';
+  '' + lib.optionalString (stdenv.hostPlatform != stdenv.targetPlatform) (
+    let
+      crossFile = writeText "cross-file.conf" ''
+        [binaries]
+        # fixes errors such as
+        # [Errno 8] Exec format error: './gdk3-scan'
```

Comment:
I think we should change the comment to mention that we want to choose the binary for the right platform when cross compiling.